include/ctype.h

```diff
@@ -37,21 +37,50 @@
 #ifndef __INCLUDE_CTYPE_H
 #define __INCLUDE_CTYPE_H
 
-/* There is no consistent ctype implementation, just a smattering of
- * functions.  Individually, they are okay, but a more standard, data lookup
- * approach would make more sense if used extensively.
- */
-
 /****************************************************************************
  * Included Files
  ****************************************************************************/
 
 #include <nuttx/compiler.h>
+#include <nuttx/arch.h>
 
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
 
+/* The classification flags */
+
+#define _U          0x01      /* upper */
+#define _L          0x02      /* lower */
+#define _N          0x04      /* digit */
+#define _S          0x08      /* space */
+#define _P          0x10      /* punct */
+#define _C          0x20      /* cntrl */
+#define _X          0x40      /* a-f|A-F */
+#define _B          0x80      /* blank */
+
+#define __ctype_lookup(c) up_romgetc(__ctype_ + (c))
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+#undef EXTERN
+#ifdef __cplusplus
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+extern const IPTR char _ctype_[];
```

Comment:
```suggestion
extern const IOBJ char _ctype_[];
```
I think that IOBJ, not IPTR is the correct qualifier here.   IOBJ indicates the storage location (__flash for AVR) and IPTR indicates memory acess method.  See Example at: https://gcc.gnu.org/onlinedocs/gcc/Named-Address-Spaces.html

Reply:
I wonder if either is needed if the ROM memory is explicitly accessed via up_romgetc().

Reply:
Done.